### PR TITLE
Create documentation

### PR DESCRIPTION
**Pull request by OkonSamuel**

### CODE REVIEW

File: docs/src/index.md [hunk 1]
```diff
@@ -1,12 +1,119 @@
...
+
+To use any model implemented in this package, the model must first be wrapped in an MLJ 
+machine alongside the required data. Users also get additional features from MLJ including 
+performance evaluation, hyper-parameter tuning, stacking etc.
+The following example shows how to train a `KNNClassifier` on the crabs dataset.
+
+```julia
+using NearestNeighborModels, MLJ
+X, y = @load_crabs; # loads the crabs dataset from MLJ
+train_inds, test_inds = partition(1:nrows(X), 0.7, shuffle=false, rng=100);
```
Comment (ablaom) [inline docs/src/index.md:36]: No point specifying `rng` if `shuffle=false` (the default). If you omit `shuffle` and specify an `rng` (integer or `AbstractRNG`) then shuffling is implied.

File: docs/src/index.md [hunk 2]
```diff
@@ -1,12 +1,119 @@
...
+
+n = 50
+p = 5
+l = 2
+rng = StableRNG(100)
+# `table` converts an `AbstractMatrix` into a `Tables.jl` compactible table
+X = table(randn(rng, (n, p))) # feature table
+Y = table(randn(rng, (n, l))) # target table
+
+train_inds, test_inds = partition(1:nrows(X), 0.8, shuffle=false, rng=rng);
```
Comment (ablaom) [inline docs/src/index.md:96]: See previous comment

**Review by ablaom — APPROVED** [review]
Okay, a simple readme documentation probably sufficed, but this is awesome! 🎉 

I'm not a super expert on Documenter generation, so there may be technical details I'm missing. It looks good to me. 

I see that you wrap all the codes in "literal" "```julia" blocks, rather than auto-generating them by using "```@example```" or "```@repl```" blocks. That's fine, if you have very carefully checked all the code blocks evaluate. Debugging the auto-generating ones can be very tedious, so I don't have a strong opinion.

**Comment by ablaom** [comment]
We will need to setup secret/deploy keys to enable deployment of the documentation onto gh pages  during CI. Let me know if you want me to set that up.

**Comment by OkonSamuel** [comment]
> I see that you wrap all the codes in "literal" `julia` blocks, rather than auto-generating them by using `@example` or `@repl` blocks (see [here](https://juliadocs.github.io/Documenter.jl/stable/man/syntax/)). That's fine, if you have very carefully checked all the code blocks evaluate. Debugging the auto-generating ones can be very tedious, so I don't have a strong opinion.

Yeah. I just add few doctests.

**Comment by OkonSamuel** [comment]
> We will need to setup secret/deploy keys to enable deployment of the documentation onto gh pages during CI. Let me know if you want me to set that up.

I have done this. But this #PR has to be merged unto dev and then unto master for the documentation to build.
` build_docs = (base_ref == "master") && (head_ref == "dev") && (base_repository == head_repository)`

File: docs/src/index.md [hunk 3]
```diff
@@ -33,37 +33,77 @@ The following example shows how to train a `KNNClassifier` on the crabs dataset.
 ```julia
 using NearestNeighborModels, MLJ
 X, y = @load_crabs; # loads the crabs dataset from MLJ
-train_inds, test_inds = partition(1:nrows(X), 0.7, shuffle=false, rng=100);
+train_inds, test_inds = partition(1:nrows(X), 0.7, shuffle=false);
```
Comment (ablaom) [inline docs/src/index.md:36]: (just FYI: You can leave `shuffle=false` out, as it's the default; no harm leaving it in for clarity)

**Pull request merged**